spikecomparison/multisortingcomparison.py

```diff
@@ -10,31 +12,120 @@
+                 do_matching=True):
         BaseComparison.__init__(self, sorting_list, name_list=name_list,
                                 delta_time=delta_time, sampling_frequency=sampling_frequency,
                                 match_score=match_score, chance_score=chance_score,
                                 n_jobs=n_jobs, verbose=verbose)
-        self._do_matching(verbose)
+        self._spiketrain_mode = spiketrain_mode
+        if do_matching:
+            self._do_comparison()
+            self._do_graph()
+            self._do_agreement()
 
     def get_sorting_list(self):
+        '''
+        Returns sorting list
+
+        Returns
+        -------
+        sorting_list: list
+            List of SortingExtractor objects
+        '''
         return self.sorting_list
 
-    def get_agreement_sorting(self, minimum_matching=0):
-        sorting = AgreementSortingExtractor(self, min_agreement=minimum_matching)
+    def get_agreement_sorting(self, minimum_agreement=1, minimum_agreement_only=False):
+        '''
+        Returns AgreementSortingExtractor with units with a 'minimum_matching' agreement.
+
+        Parameters
+        ----------
+        minimum_agreement: int
+            Minimum number of matches among sorters to include a unit.
+        minimum_agreement_only: bool
+            If True, only units with agreement == 'minimum_matching' are included.
+            If False, units with an agreement >= 'minimum_matching' are included
+
+        Returns
+        -------
+        agreement_sorting: AgreementSortingExtractor
+            The output AgreementSortingExtractor
+        '''
+        assert minimum_agreement > 0, "'minimum_agreement' should be greater than 0"
+        sorting = AgreementSortingExtractor(self, min_agreement=minimum_agreement,
+                                            min_agreement_only=minimum_agreement_only)
         sorting.set_sampling_frequency(self.sampling_frequency)
         return sorting
 
-    def _do_matching(self, verbose):
+    def compute_subgraphs(self):
+        '''
+        Computes subgraphs of connected components.
+
+        Returns
+        -------
+        sg_sorter_names: list
+            List of sorter names for each node in the connected component subrgaph
+        sg_units: list
+            List of unit ids for each node in the connected component subrgaph
+        '''
+        g = self.graph
+        subgraphs = (g.subgraph(c).copy() for c in nx.connected_components(g))
+        sg_sorter_names = []
+        sg_units = []
+        for i, sg in enumerate(subgraphs):
+            sorter_names = []
+            sorter_units = []
+            for node in sg.nodes:
+                sorter_names.append(node.split('_')[0])
+                sorter_units.append(int(node.split('_')[1]))
+            sg_sorter_names.append(sorter_names)
+            sg_units.append(sorter_units)
+        return sg_sorter_names, sg_units
+
+    def dump(self, save_folder):
```

Comment:
Good idea, I plan to do it for other comparison for long time.
If i find soe time, we could move this at base class.